unstructured/partition/epub.py

```diff
@@ -29,13 +37,47 @@ def partition_epub(
         If True, the output will include page breaks if the filetype supports it
     metadata_date
         The last modified date for the document.
-
     """
-    return convert_and_partition_html(
-        source_format="epub",
-        filename=filename,
-        file=file,
-        include_page_breaks=include_page_breaks,
-        metadata_filename=metadata_filename,
-        metadata_date=metadata_date,
-    )
+    exactly_one(filename=filename, file=file)
+    if filename is None:
+        filename = ""
+
+    if file is not None:
+        with tempfile.NamedTemporaryFile(delete=False) as tmp:
+            tmp.write(file.read())
+            filename = tmp.name
+        last_modification_date = get_last_modified_date_from_file(file)
+    else:
+        last_modification_date = get_last_modified_date(filename)
+
+    book = epub.read_epub(filename)
+    toc_items = list(book.toc)
+    elements = []
+
+    for toc_item in toc_items:
+        # Some toc items may be tuple
+        if isinstance(toc_item, tuple):
+            toc_item = toc_item[0]
+
+        href = toc_item.href.split("#")[0]
+        title = toc_item.title
+        item = book.get_item_with_href(href)
+
+        if item is not None:
+            try:
+                # Convert the item content to a string
+                html_content = item.get_content().decode()
+                section_elements = partition_html(
+                    text=html_content,
+                    epub_section=title,
+                    metadata_date=metadata_date,
+                    last_modification_date=metadata_date or last_modification_date,
+                    **kwargs,
+                )
+            except Exception as e:
```

Comment:
We shouldn't have a bare `except` here. Are you hitting an error that you need to handle when testing this?

Reply:
I don't remember off the top of my head, but I don't believe it was raising an error. I'll remove it.